gcloud/storage/_helpers.py

```diff
@@ -187,3 +190,14 @@ def _setter(self, value):
         self._patch_properties({fieldname: value})
 
     return property(_getter, _setter)
+
+
+def _base64_md5hash(bytes_to_sign):
```

Comment:
Wouldn't this be better off taking a fileobj, and then loop over reading chunks?  E.g.:

```python 
def _base64_md5hash(fileobj_to_sign):
    """Get MD5 hash of bytes (as base64).
    
    :type fileobj_to_sign: readable file-like object
    :param fileobj_to_sign: Stream used to compute an MD5 hash (as base64).
    """
    hash = MD5.new()
    while True:
        chunk = fileobj_to_sign.read(CHUNKSIZE)
        if chunk:
            hash.update(chunk)
        else:
            break
    digest_bytes = hash.digest()
    return base64.b64encode(digest_bytes)
```